docs/about/release-notes.rst

```diff
@@ -18,9 +18,9 @@ Breaking changes
 ~~~~~~~~~~~~~~~~
```

Comment:
In addition, I think that the `affine3` dtype listed above (https://github.com/scipp/scipp/blob/main/docs/about/release-notes.rst?plain=1#L12) does not exist. I think it should be `affine_transform3` instead?